src/Parser.js

```diff
@@ -157,7 +157,7 @@ export function ruleset (value, root, index, offset, rules, points, type, props,
  * @return {object}
  */
 export function comment (value, root) {
-	return node(value, root, COMMENT, from(char()), substr(value, 2, -2), 0)
+	return node(value, root, COMMENT, from(char()), root, substr(value, 2, -2), 0)
```

Comment:
Why are we re-using the root reference, i'd assume we'd need to pass a new parameter to these functions: `parent`?

Reply:
As far as I can tell - comment and declaration node points to their parent from the `.root` property, so this is "only" providing consistency with rule nodes (so each get non-null `.parent` when appropriate)

Reply:
oh, i missed the fact that they where the `comment/declaration` functions.

Reply:
When thinking about this more I thought I would have to adjust the code for those functions to accept `parent` after all as in:
```
compile(`a{@media{w:1;}}`)
```
the parent of this declaration should be `@media`. So I went back checking things and it turns out that the `.parent` is `@media` for this - because it's also the `.root`, which in turn leads me to a question - should the `.root` of this declaration be `@media` or `.children[0]` of this `@media` node? I thought it's the latter, but maybe I'm just misunderstanding the exact purpose of `.root`?

Reply:
It is for ruleset nodes, the `.root` for `decl`/`comm` is as it should be, i.e it will also be `@viewport` for `@viewport{width:device-width;}`.

Reply:
To what declaration's `.root` should point here?
```js
compile(`.user{@media{@supports{b:1}}}`)
```